src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,141 @@
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update " + car + " in DB.", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    private List<Driver> getDriversForCar(Long id) {
+        String query = "SELECT id, name, license_number "
+                + "FROM drivers d "
+                + "JOIN cars_drivers cd ON cd.driver_id = d.id AND cd.car_id = ?";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection driverConnection = ConnectionUtil.getConnection();
+                PreparedStatement driverStatement = driverConnection.prepareStatement(query)) {
+            driverStatement.setLong(1, id);
+            ResultSet resultSet = driverStatement.executeQuery();
+            while (resultSet.next()) {
+                Long driverId = resultSet.getObject("id", Long.class);
+                String name = resultSet.getString("name");
+                String licenseNumber = resultSet.getString("license_number");
+                Driver driver = new Driver(driverId, name, licenseNumber);
+                drivers.add(driver);
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of drivers from DB.", e);
+        }
+    }
+
+    private Car parseCarFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String manufacturerName = resultSet.getString("name");
+        String manufacturerCountry = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(manufacturerId, manufacturerName,
+                manufacturerCountry);
+        List<Driver> drivers = getDriversForCar(id);
```

Comment:
You opened Connection in Connection : 
1 - you called method get() and opened first Connection
2 - in method get() you called method parseCarFromResultSet()
3 - and in method parseCarFromResultSet() you called method getDriversForCar()
4 - and in method getDriversForCar() you called SECOND CONNECTION , but you HADN'T CLOSED FIRST YET.

The same thing with getAll() method

Try to fix it